program/tests/tests/unstake.rs

```diff
@@ -0,0 +1,123 @@
+// SPDX-FileCopyrightText: 2021 Chorus One AG
+// SPDX-License-Identifier: GPL-3.0
+
+#![cfg(feature = "test-bpf")]
+
+use crate::assert_solido_error;
+use crate::context::{Context, StakeDeposit};
+use lido::MINIMUM_STAKE_ACCOUNT_BALANCE;
+use lido::{error::LidoError, token::Lamports};
+use solana_program::stake::state::StakeState;
+use solana_program_test::tokio;
+use solana_sdk::signer::Signer;
+const STAKE_AMOUNT: Lamports = Lamports(10_000_000_000);
+
+struct UnstakeContext {
+    context: Context,
+}
+
+impl UnstakeContext {
+    async fn new(stake_amount: Lamports) -> UnstakeContext {
```

Comment:
`UnstakeContext` has a single field and only a constructor, we can simplify this to just a function

```rust
async fn new_unstake_context(stake_amount: Lamports) -> Context
```